src/SenseiConfig.hs

```diff
@@ -0,0 +1,82 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module SenseiConfig where
+
+import GHC.Generics
+import Data.Aeson
+import Data.Yaml
+import System.Directory
+import System.Process
+import System.Exit
+import Text.Casing
+
+senseiGlobalConfigFilename :: FilePath
+senseiGlobalConfigFilename = "senseirc"
+
+senseiLocalConfigFilename :: FilePath
+senseiLocalConfigFilename = ".senseirc"
+
+data SenseiConfig = SenseiConfig {
+  -- | Shell command to run after a file has changed, but before reloading. Cancels the current
+  -- trigger cycle when exiting with non-zero exit code.
+  senseiConfigBeforeReload :: Maybe String
```

Comment:
If we stick with camelCase then I think it could make sense to name the fields exactly the same way as the config keys so that you can derive `FromJSON`.

If we don't export `SenseiConfig` (see my other comments) then it doesn't really matter how the fields are named. Renaming `SenseiConfig` to `ConfigSyntax` could also be nice.